module/functions/Get-ExtensionDependencies.Tests.ps1

```diff
@@ -0,0 +1,273 @@
+                                @{
+                                    Name = "an-extension"
+                                    Version = "1.0.0"
+                                }
+                                @{
+                                    Name = "another-extension"
+                                    Version = "2.0.0"
+                                }
+                                @{
+                                    Name = "an-extension-from-git"
+                                    GitRepository = "https://github.com/myOrg/myExtension"
+                                    GitRef = "main"
+                                }
+                            )
+                        }
+                    }
+                }
+                Mock Import-PowerShellDataFile { $mockModuleManifest }
+                [array]$deps = Get-ExtensionDependencies $extensionConfig
+            }
+            AfterAll {}
+
+            It 'Should resolve all the dependencies' {
+                $deps.Count | Should -Be 3
+            }
+            It 'Should return the correct dependency metadata' {
+                $deps[0].Name | Should -Be 'an-extension'
+                $deps[0].Version | Should -Be '1.0.0'
+                $deps[1].Name | Should -Be 'another-extension'
+                $deps[1].Version | Should -Be '2.0.0'
+                $deps[2].Name | Should -Be 'an-extension-from-git'
+                $deps[2].GitRepository | Should -Be 'https://github.com/myOrg/myExtension'
+                $deps[2].GitRef | Should -Be 'main'
+            }
+        }
+
+        Context 'Module manifest private data definition - mixed syntax' {
+            BeforeAll {
+                $mockModuleManifest = @{
+                    RootModule = "an-extension"
+                    PrivateData = @{
+                        PSData = @{
+                            ExternalModuleDependencies = @()
+                        }
+                        ZeroFailed = @{
+                            ExtensionDependencies = @(
+                                "an-extension"
+                                @{
+                                    Name = "another-extension"
+                                    Version = "2.0.0"
+                                }
+                            )
+                        }
+                    }
+                }
+                Mock Import-PowerShellDataFile { $mockModuleManifest }
+                [array]$deps = Get-ExtensionDependencies $extensionConfig
+            }
+            AfterAll {}
+
+            It 'Should resolve all the dependencies' {
+                $deps.Count | Should -Be 2
+            }
+            It 'Should return the correct dependency metadata' {
+                $deps[0].Name | Should -Be 'an-extension'
+                $deps[1].Name | Should -Be 'another-extension'
+                $deps[1].Version | Should -Be '2.0.0'
+            }
+        }
+    }
+}
```

Comment:
Do you need to think about validation logic for what data types are allowed in each value - i.e. pick up typos etc?
What about verifying diagnostic logging output to help you troubleshoot in a DevOps pipeline?